applications/FemToDemApplication/python_scripts/CouplingFemDem.py

```diff
@@ -600,31 +617,30 @@ def GetNodeCoordinates(self, Node):
 #============================================================================================================================
     def UpdateDEMVariables(self):
 
-        DEM_Nodes = self.SpheresModelPart.Nodes
-
-        for DEM_Node in DEM_Nodes:  # Loop over DEM nodes
-            if (DEM_Node.GetValue(KratosFemDem.INACTIVE_NODE) == False):
+        FEM_Nodes = self.FEM_Solution.main_model_part.Nodes
+        for fem_node in FEM_Nodes:
+            if fem_node.GetValue(KratosFemDem.IS_DEM):
+                id_node = fem_node.Id
+                associated_dem = self.SpheresModelPart.GetNode(id_node)
 
-                Id = DEM_Node.Id
-                Corresponding_FEM_Node = self.FEM_Solution.main_model_part.GetNode(Id)
-                Coordinates    = self.GetNodeCoordinates(Corresponding_FEM_Node)
-                Velocity_x     = Corresponding_FEM_Node.GetSolutionStepValue(KratosMultiphysics.VELOCITY_X)
-                Velocity_y     = Corresponding_FEM_Node.GetSolutionStepValue(KratosMultiphysics.VELOCITY_Y)
-                Displacement_x = Corresponding_FEM_Node.GetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_X)
-                Displacement_y = Corresponding_FEM_Node.GetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Y)
+                Coordinates    = self.GetNodeCoordinates(fem_node)
+                Velocity_x     = fem_node.GetSolutionStepValue(KratosMultiphysics.VELOCITY_X)
```

Comment:
Better GetSolutionStepValue(VELOCITY), then assign then components of the vector []

Reply:
Otherwise you are accessing to the database several times

Reply:
good point, done